### PR TITLE
Fix 1.29 tests by pinning newly-incompatible log crate

### PR DESCRIPTION
**Pull request by sgeisler**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -53,9 +53,11 @@ Because some dependencies have broken the build in minor/patch releases, to
 compile with 1.29.0 you will need to run the following version-pinning command:
 ```
 cargo update --package "cc" --precise "1.0.41"
+cargo update --package "log:0.4.x" --precise "0.4.13" # x being the highest patch version
 cargo update --package "cfg-if" --precise "0.1.9"
 cargo update --package "unicode-normalization" --precise "0.1.9"
 cargo update --package "serde_json" --precise "1.0.39"
 cargo update --package "serde" --precise "1.0.98"
 cargo update --package "serde_derive" --precise "1.0.98"
+cargo update --package "byteorder" --precise "1.3.4"
```
Comment (sgeisler) [inline README.md:62]: I just fixed the missing documentation of behavior already present in tests in the same commit. I can split it off if you want (just noticed after opening the PR).

File: contrib/test.sh [hunk 2]
```diff
@@ -11,6 +11,11 @@ echo "PATH: \"$PATH\""
 # Pin dependencies for Rust v1.29
 if [ "$TRAVIS_RUST_VERSION" = "1.29.0" ]; then
     cargo generate-lockfile --verbose
+
+    # hyper depends on log 0.3 while we depnd on 0.4, so cargo doesn't know which one to pin
+    LOG_4_PATCH="$(cargo update --package "log" --precise "0.4.13" 2>&1 | sed -n "s/.*log:0.4.\([0-9]*\)/\1/p")"
```
Comment (sgeisler) [inline contrib/test.sh:16]: I couldn't find a good way to get the log versions so I just ran the command in a way that it fails, giving me the version number to run it correctly.

**Comment by stevenroose** [comment]
Oh I think I just published a similar PR (going through e-mail anti-chronologically for once, should not do that :|).

**Comment by stevenroose** [comment]
Ah yeah I also ran into your problem. You need to be able to build before you can name the dependencies by their version but with 1.29 you can't build so you can't use the pin command, it's basically a Cargo bug.

I know limiting the version in Cargo.toml is not really a good practice, but for now https://github.com/rust-bitcoin/rust-bitcoincore-rpc/pull/168 works...

File: contrib/test.sh (hunk 2 above)
Comment (stevenroose) [reply contrib/test.sh:16]: Hmm having added that error in a comment would have been nice, but let's see if this works. I prefer it over the version setting in Cargo.toml.

**Review by stevenroose — APPROVED** [review]

**Pull request merged**